awscli/data/ec2/2016-11-15/completions-1.json

```diff
@@ -1701,6 +1701,781 @@
         ]
       }
     },
+    "DescribeAvailabilityZones": {
+      "ZoneNames": {
+        "completions": [
+          {
+            "parameters": {},
+            "resourceName": "AvailabilityZone",
+            "resourceIdentifier": "ZoneName"
+          }
+        ]
+      }
+    },
+    "DescribeClassicLinkInstances": {
+      "InstanceIds": {
+        "completions": [
+          {
+            "parameters": {},
+            "resourceName": "InstanceCreditSpecification",
```

Comment:
Yeah for a number of these changes, I don't think the existing approach is going to work, even with the modifications I've made.  The issue seems to be in the reverse mapping of resourceIdentifiers back to resource, and specifically that we can't assume that a resourceIdentifier for a resource (such as InstanceId) uniquely maps back to a single resource (Instance).  We need a smarter way to match up `DescribeClassicLinkInstances` to the appropriate resource type.  I'll have to think about how to best do this.

Reply:
Other thing to note, for this specific case one of the complications is that getting instance ids requires a `Reservations[].Instances[].InstanceId`, and we only go one level deep in the response structure.  We assume the identifier is a member of the structure of the associated list member in the response, e.g `Instances[].InstanceId`.

I'm actually thinking that ec2 might just need some handwritten overrides rather than trying to make the heuristics more sophisticated...